sklearn/preprocessing/data.py

```diff
@@ -788,7 +788,7 @@ def transform(self, X, y=None):
             warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
 
         if sparse.issparse(X):
-            if X.shape[0] == 1:
+            if X.shape[1] != self.scale_.shape[0]:
```

Comment:
Transforming on an input sparse matrix with shape  `(1, n)` should return something with shape `(1, n)` (1 transformed sample with n features) without any error. Each feature should be scaled by the matching entry in the  `scale_` array.